pkg/providers/instance/offerings/commonerrorhandlers_test.go

```diff
@@ -0,0 +1,197 @@
+/*
+Portions Copyright (c) Microsoft Corporation.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package offerings
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/karpenter-provider-azure/pkg/apis/v1beta1"
+	"github.com/Azure/karpenter-provider-azure/pkg/cache"
+	"github.com/Azure/skewer"
+	"github.com/stretchr/testify/assert"
```

Comment:
pet peeve of mine, but we should use a single assertion library. I'm fine with some tests being Ginkgo and some not (I prefer not but at least for E2E tests I see the value) - but IMO we should just use Ginkgo for asserts in both.

Otherwise you get a mix of assertion styles and it becomes more confusing (IMO) to read the code and the test output.

I know we use testify some elsewhere, but only in 16 files compared to 69 files for Gomega (many of those are also Ginkgo, but not all).

Let's unify on Gomega for assertions regardless of Ginkgo or not?

Reply:
Ahh you didn't actually change this (just copied it)

Reply:
My point is more about assertion libraries than test formats, tohugh I agree both are worth talking about.

I'm fine with:
1. Gomega (asserts) in go-test
2. Gomega (asserts) in Ginkgo (test format)

I'm less fine with
1. Gomega (asserts) in go-test
2. testify (asserts) in go-test
3. Gomega (asserts) in Ginkgo

So IMO it's not 1 vs 2, it's 2 vs 3.

Reply:
agree you dont need to fix here though.